apps/app/src/views/routes/Settings/components/users/EditRolesAction.tsx

```diff
@@ -0,0 +1,444 @@
+import React from 'react';
+import {
+  HStack,
+  ModalContent,
+  ModalHeader,
+  ModalBody,
+  Alert,
+  AlertIcon,
+  VStack,
+  Text,
+  Box,
+  Button,
+  ModalFooter,
+  FormControl,
+  FormLabel,
+  FormErrorMessage,
+  Input
+} from '@chakra-ui/react';
+import * as yup from 'yup';
+import { rolesSchema, RolesSelect } from '../utils/Roles';
+import { useMutation, useQuery } from '@apollo/client';
+import { useClinicalApiClient } from '../../apollo';
+import { graphql } from 'apps/app/src/gql';
+import { FormikStateSelect, yupStateSchema } from '../utils/States';
+import { FormikTouched, FormikErrors, ErrorMessage, Field, Formik } from 'formik';
+
+interface EditRolesActionProps {
+  userId: string;
+  onClose: () => void;
+}
+
+const GetUserQuery = graphql(/* GraphQL */ `
+  query Query($userId: ID!) {
+    user(id: $userId) {
+      address {
+        street1
+        street2
+        state
+        postalCode
+        country
+        city
+      }
+      npi
+      phone
+      name {
+        first
+        full
+        last
+        middle
+        title
+      }
+      fax
+      email
+    }
+  }
+`);
+
+const SetUserRolesMutation = graphql(/* GraphQL */ `
+  mutation SetUserRoles($userId: ID!, $roles: [ID!]!) {
+    setUserRoles(userId: $userId, roles: $roles)
+  }
+`);
+
+const UpdateProviderProfileMutation = graphql(/* GraphQL */ `
+  mutation UpdateProviderProfile(
+    $providerId: ID!
+    $updateProviderProfileInput: UpdateProviderProfileInput!
+  ) {
+    updateProviderProfile(providerId: $providerId, input: $updateProviderProfileInput)
+  }
+`);
+
+const hasPrescriberRole = (roles: { value: string; label: string }[]) =>
+  roles.some((r) => r.label === 'Prescriber');
+
+const roleSchema = yup
+  .object({
+    email: yup
+      .string()
+      .email('Invalid email')
+      .required("Please include email of the person you'd like to invite"),
+    roles: rolesSchema.required().min(1, 'Must have at least one role'),
+    provider: yup
+      .object({
+        npi: yup
+          .string()
+          .required('NPI is required for prescribers')
+          .matches(/^[0-9]*$/, { message: 'Invalid NPI' }),
```

Comment:
this regex will match the empty string.

Reply:
This was grabbed from Invite flow. Changed to avoid acceptance of empty string.